src/cdomain/value/cdomains/intDomain.ml

```diff
@@ -668,21 +679,9 @@ struct
     | Some (l0,u0), Some (l1,u1) ->
       let (min_ik, max_ik) = range ik in
       let threshold = get_interval_threshold_widening () in
-      let upper_threshold u =
-        let ts = if get_interval_threshold_widening_constants () = "comparisons" then WideningThresholds.upper_thresholds () else ResettableLazy.force widening_thresholds in
-        let u = Ints_t.to_bigint u in
-        let t = List.find_opt (fun x -> Z.compare u x <= 0) ts in
-        BatOption.map_default Ints_t.of_bigint max_ik t
-      in
-      let lower_threshold l =
-        let ts = if get_interval_threshold_widening_constants () = "comparisons" then WideningThresholds.lower_thresholds () else ResettableLazy.force widening_thresholds_desc in
-        let l = Ints_t.to_bigint l in
-        let t = List.find_opt (fun x -> Z.compare l x >= 0) ts in
-        BatOption.map_default Ints_t.of_bigint min_ik t
-      in
-      let lt = if threshold then lower_threshold l1 else min_ik in
+      let lt = if threshold then IArith.lower_threshold l1 min_ik else min_ik in
       let l2 = if Ints_t.compare l0 l1 = 0 then l0 else Ints_t.min l1 (Ints_t.max lt min_ik) in
-      let ut = if threshold then upper_threshold u1 else max_ik in
+      let ut = if threshold then IArith.upper_threshold u1 max_ik else max_ik in
       let u2 = if Ints_t.compare u0 u1 = 0 then u0 else Ints_t.max u1 (Ints_t.min ut max_ik) in
```

Comment:
The reason non-set intervals didn't have this problem is because if the `Ints_t.min ut max_ik` here to make sure that if `ut` was chosen to be bigger than the type range, then `max_ik` is used instead. Interval sets probably didn't have such check.
But now it should be unnecessary because `IArith.upper_threshold` should ensure that it doesn't return anything bigger.

I'm also confused by the `Ints_t.max u1` here. It shouldn't be possible for the threshold/`max_ik` to be smaller than `u1` to begin with!?

On a different note, currently `ut` is computed even if `u2` doesn't need it, which is wasteful. We could just move `ut` into the else branch to avoid that.